File: app/src/main/java/com/sagoforest/template/business/interfaces/usecases/IUsersUseCase.java
package com.sagoforest.template.business.interfaces.usecases;

import com.sagoforest.template.da.entities.User;

import java.util.List;

import io.reactivex.Flowable;

/**
 * Use case that manages users
 * <p>
 * Created by andy on 1/30/18.
 */

public interface IUsersUseCase {

    Flowable<List<User>> getUsers();

    void addNew(String firstName, String lastName);

    void clearAll();
}

File: app/src/main/java/com/sagoforest/template/business/interfaces/usecases/IUsersUseCase.kt
package com.sagoforest.template.business.interfaces.usecases

import com.sagoforest.template.da.entities.User

import io.reactivex.Flowable

/**
 * Use case that manages users
 *
 *
 * Created by andy on 1/30/18.
 */

interface IUsersUseCase {

    val users: Flowable<List<User>>

    fun addNew(firstName: String, lastName: String)

    fun clearAll()
}
